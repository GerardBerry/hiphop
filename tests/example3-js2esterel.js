var js2esterel = require("../js2esterel.js");
var example = require("./example3.js");

console.log(example.machine.esterel_code());
