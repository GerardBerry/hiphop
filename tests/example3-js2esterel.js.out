example3> ;
--- Output: T
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> A;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output: T
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
abort
    loop
        emit S;
        if S then
            emit T
        end if;
        pause;
        emit V
    end loop
when A
